1971.py

```diff
@@ -0,0 +1,48 @@
+# class Solution:
+#     def validPath(self, n: int, edges: List[List[int]], start: int, end: int) -> bool:
+#         seen, graph = set(), defaultdict(list)
+
+#         for s, e in edges:
+#             graph[s].append(e)
+#             graph[e].append(s)
+
+#         queue = deque([start])
+#         seen.add(start)
+
+#         while queue:
+#             curr = queue.popleft()
+
+#             if curr == end:
+#                 return True
+
+#             for v in graph[curr]:
+#                 if v not in seen:
+#                     queue.append(v)
+#                     seen.add(v)
+
+#         return False
+
+class Solution:
+    def validPath(self, n: int, edges: List[List[int]], start: int, end: int) -> bool:
+        seen, graph = set(), defaultdict(list)
```

Comment:
The logic of this piece of code is correct. However, it seems that you are still confusing yourself with `seen` and `visited`. 

In BFS, we maintain a set of nodes in the `seen` variable because these nodes have just been "seen" and not "visited" yet. 

Here in DFS, when you add the node to the set (in line 40), you have already "visited" that node (the current `dfs()` recursive call belongs to the `node` you passed in). Therefore, naming it to `visited` is more appropriate and should help you understand the subtle difference between DFS and BFS when it comes handling the "states" of the nodes.